### DISCUSSION TITLE
Import PPO back to AnyLogic

### DISCUSSION THREAD

**Discussion by soheile-antar** [the-anylogic-company/Alpyne#25] [Q&A]
Hello.
I was testing the stock management example. I prepared the RLpolicy and I got the PPO file. Any one knows how I can import this PPO back to AnyLogic to use the results there?

**Comment by t-wolfeadam** [comment]
If you have statistics tracking variables as part of your observation, you can still use alpyne for testing purposes. In case you want to explore it deeper or more visually, here are a few ways you can reimport it:

- Easiest way: [Pypeline ](https://github.com/t-wolfeadam/AnyLogic-Pypeline)- this way you can still keep the policy related logic in Python; the downside is overall execution speed will be slower
- More involved but still simple way: ONNX - if you can convert it to a .onnx file, you can use [my helper library](https://github.com/t-wolfeadam/AnyLogic-ONNX-Helper) for easily querying the policy without sacrificing speed
- Harder but most flexible way: A native Java library that can query your policy file. This will vary depending on your export type.
